File: src/scripts/create.js
const { error } = require('console');
const fs = require('fs');
const exec = require('child_process').exec;
const cmpt = process.argv[2];
fs.readFile('./src/components/template.html', 'utf8', (err, source) =>{
        if (err) return console.error(err);
        const content = source.replace(/COMPONENT_NAME/g, cmpt);
        if (fs.existsSync(`./src/components/${cmpt}.html`)) {
            return console.error(`${cmpt}.html already exist, use another name`);
        }
        fs.writeFile(`./src/components/${cmpt}.html`, content, (err) => {
            if (err) return console.error(`there is a problem in creating ${cmpt}.html`);
            else{
                fs.writeFile(`./src/assets/sass/components/${cmpt}.scss`, '', (err)=>{
                    if (err) return console.error(`there is a problme in creating ${cmpt}.sass`);
                    console.log(`${cmpt} created successfully!`);
                    exec(`code -r ./src/components/${cmpt}.html` , (err)=>{
                        if (err) return console.error(err);
                    });
                    exec(`code -r ./src/assets/sass/coponents/${cmpt}.scss` , (err)=>{
                        if (err) return console.error(err);
                    });
                });
            }
        });
});

File: src/scripts/create.py
import os
import logging
import sys 

def create_component(component_name):
  """Creates a new component with HTML and SCSS files.

  Args:
    component_name: The name of the component to create.
  """
  template_path = os.path.join("src", "components", "template.html")
  component_html_path = os.path.join("src", "components", f"{component_name}.html")
  component_scss_path = os.path.join("src", "assets", "sass", f"{component_name}.scss")

  # Configure logging (optional)
  logging.basicConfig(filename='component_creator.log', level=logging.DEBUG)

  # Read the template file
  try:
    with open(template_path, "r", encoding="utf-8") as f:
      source = f.read()
  except FileNotFoundError:
    logging.error(f"Template file not found: {template_path}")
    return

  # Replace placeholder with component name
  content = source.replace("COMPONENT_NAME", component_name)

  # Check if component HTML already exists
  if os.path.exists(component_html_path):
    logging.error(f"{component_name}.html already exists, use another name")
    return

  # Create component HTML file
  try:
    with open(component_html_path, "w", encoding="utf-8") as f:
      f.write(content)
  except OSError as e:
    logging.error(f"Error creating {component_name}.html: {e}")
    return

  # Create empty component SCSS file
  try:
    with open(component_scss_path, "w", encoding="utf-8") as f:
      f.write("")
  except OSError as e:
    logging.error(f"Error creating {component_name}.scss: {e}")
    return

  # Open files in VS Code (optional)
  try:
    os.system(f"code -r {component_html_path}")
    os.system(f"code -r {component_scss_path}")
  except OSError as e:
    logging.error(f"Error opening files in VS Code: {e}")

  logging.info(f"{component_name} created successfully!")

# Get component name from command line argument (if provided)
if len(sys.argv) > 2:
  component_name = sys.argv[2]
  create_component(component_name)
else:
  logging.error("Please provide a component name as an argument.")
